community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/slurmsync.py

```diff
@@ -470,6 +472,92 @@ def update_topology(lkp: util.Lookup) -> None:
         log.debug("Topology configuration updated. Reconfiguring Slurm.")
         util.scontrol_reconfigure(lkp)
 
+
+def delete_reservation(lkp: util.Lookup, reservation_name: str) -> None:
+    util.run(f"{lkp.scontrol} delete reservation {reservation_name}")
+
+
+def create_reservation(lkp: util.Lookup, reservation_name: str, node: str, start_time: datetime) -> None:
+    # Format time to be compatible with slurm reservation.
+    formatted_start_time = start_time.strftime('%Y-%m-%dT%H:%M:%S')
+    util.run(f"{lkp.scontrol} create reservation user=slurm starttime={formatted_start_time} duration=180 nodes={node} reservationname={reservation_name}")
+
+
+def get_slurm_reservation_maintenance(lkp: util.Lookup) -> Dict[str, datetime]:
+    res = util.run(f"{lkp.scontrol} show reservation --json")
+    all_reservations = json.loads(res.stdout)
+    reservation_map = {}
+
+    for reservation in all_reservations['reservations']:
+        reservation_name = reservation['name']
+        nodes = reservation['node_list']
+        time = datetime.fromtimestamp(reservation['start_time']['number'])
+
+        if reservation_name is None or nodes is None or time is None:
```

Comment:
* `time` would neve be a `None`, 
In case`reservation` doesn't have have `start_time` or `reservation['start_time']` doesn't have `number` function will fail with `KeyError`. (Same goes for `reservation[name]`)

If you want to do it in "None-safe" way, write it like this:

```py
name = reservation.get('name')
nodes = reservation.get('nodes')
time_epoch = reservation.get('start_time', {}).get('number')

 if name is None or nodes is None or time_epoch is None:
    continue
```

* `time` is the name of standard library, please use other name, e.g. `start_time`.